src/AspNetCore/Src/forwardinghandler.cxx

```diff
@@ -3147,9 +3147,10 @@ FORWARDING_HANDLER::TerminateRequest(
 {
     if (m_pApplication->QueryConfig()->QueryHostingModel() == HOSTING_IN_PROCESS) 
     {
-        //
-        // Todo: need inform managed layer to abort the request
-        //
+        // TODO Terminate Request is being called on Shutdown of IISExpress, causing 
+        // disconnect to be called at the wrong time.
+        IN_PROCESS_APPLICATION* application = (IN_PROCESS_APPLICATION*)m_pApplication;
```

Comment:
Can this fire after the request is over?

Reply:
I don't believe so now. With this change: https://github.com/aspnet/AspNetCoreModule/pull/227/files/5dd3a7777d1d021683c447532dcc0868032a9dc5#diff-6752c6fb79cdaf9a4ff73a601b7b4e5fR93, we now will unregister the disconnect context from the request.